spinta/manifests/open_api/helpers.py

```diff
@@ -2,99 +2,325 @@
 
 import json
 import re
+from collections.abc import Generator
 from pathlib import Path
-from typing import Generator
 
 from spinta.core.ufuncs import Expr
-from spinta.utils.naming import to_dataset_name, to_code_name
+from spinta.utils.naming import Deduplicator, to_code_name, to_dataset_name, to_model_name, to_property_name
 
-
-SUPPORTED_PARAMETER_LOCATIONS = {'query', 'header', 'path'}
-DEFAULT_DATASET_NAME = 'default'
+SUPPORTED_PARAMETER_LOCATIONS = {"query", "header", "path"}
+DEFAULT_DATASET_NAME = "default"
 
 
 def replace_url_parameters(endpoint: str) -> str:
     """Replaces parameters in given endpoint to their codenames.
 
     e.g. /api/cities/{cityId}/ -> /api/cities/{city_id}
     """
-    return re.sub(r'{([^{}]+)}', lambda match: f'{{{to_code_name(match.group(1))}}}', endpoint)
+    return re.sub(r"{([^{}]+)}", lambda match: f"{{{to_code_name(match.group(1))}}}", endpoint)
 
 
 def read_file_data_and_transform_to_json(path: Path) -> dict:
     with Path(path).open() as file:
         return json.load(file)
 
 
+def resolve_refs(schema: dict, root: dict) -> dict:
+    if isinstance(schema, dict):
+        if "$ref" in schema:
+            ref_path = schema["$ref"].lstrip("#/").split("/")
+            resolved = root
+            for part in ref_path:
+                resolved = resolved[part]
+            return resolve_refs(resolved, root)
+
+        return {key: resolve_refs(value, root) for key, value in schema.items()}
+
+    return schema
+
+
 def get_namespace_schema(info: dict, title: str, dataset_prefix: str) -> Generator[tuple[None, dict], None, None]:
-    yield None, {
-        'type': 'ns',
-        'name': dataset_prefix,
-        'title': info.get('summary', title),
-        'description': info.get('description', ''),
-    }
+    yield (
+        None,
+        {
+            "type": "ns",
+            "name": dataset_prefix,
+            "title": info.get("summary", title),
+            "description": info.get("description", ""),
+        },
+    )
 
 
 def get_resource_parameters(parameters: list[dict]) -> dict[str, dict]:
     resource_parameters = {}
     for index, value in enumerate(parameters):
-        name = value['name']
-        location = value['in'] if value['in'] in SUPPORTED_PARAMETER_LOCATIONS else ''
-        resource_parameters[f'parameter_{index}'] = {
-            'name': to_code_name(name),
-            'source': [name],
-            'prepare': [Expr(location)],
-            'type': 'param',
-            'description': value.get('description', ''),
+        name = value["name"]
+        location = value["in"] if value["in"] in SUPPORTED_PARAMETER_LOCATIONS else ""
+        resource_parameters[f"parameter_{index}"] = {
+            "name": to_code_name(name),
+            "source": [name],
+            "prepare": [Expr(location)],
+            "type": "param",
+            "description": value.get("description", ""),
```

Comment:
Ok this time, but let's avoid formatting old changes with ruff/black, the changes get mixed up and it's harder to review the PR this way.

Best way is to:
- Do the ruff/black refactor before you touch a file, merge that into main/master (since it should not break anything related, and if it does, tests should inform about it)
- Start making your changes on the refactored code